qxf2/newsletter_automation
# 21 jan 2022 sprint article editor notification

This PR is for an enhancement to the newsletter-generator app to have the article editor tracked in the app. 

Context: Whenever any link gets posted on Skype, it goes to newsletter-generator and an employee is assigned who needs to add description to the article. While the article editor name is currently displayed in Skype, with this change the name will also be tracked in the app. 

Changes made:
-> new column article_editor has been added to handle the name of the person that has been assigned to the article
-> the code for picking the article editor is present in another lambda which is now added to the staging-newsletter-url-filter lambda

How to test:
1. Get setup with the newsletter-generator app locally. 
Tip: Can use docker - <more details here> to setup MySQL running locally on 3306 port
The SSO was bypassed temporarily for local setup

2. Start the flask app  
python newsletter_automation/run.py

3. Run the DB migration steps, refer the README.md file for instructions
(if you hit `Target database is not up to date` error during BD migration, follow the steps in [here](https://stackoverflow.com/a/58586211) )

4. Send a request 
In a python interpreter try:
import requests
url = 'http://127.0.0.1:5000/api/articles'
data = {'url': 'https://qxf2.com', 'category_id': 2, 'article_editor': 'team3'}
response = requests.post(url=url, data=data)

Note: The 'url' needs to be unique for each request 

5. Check 'http://127.0.0.1:5000/manage-articles' to see the data that has been added
